test/db.tests.js

```diff
@@ -640,16 +674,18 @@ describe('LimitDBRedis', () => {
     });
 
     it('should work with negative values', (done) => {
```

Comment:
imho, it's a little bit weird that a bucket can have a negative value, but I'm not changing that behavior. I think I've followed the spirit of the existing semantics here, where subtracting 100 tokens from a bucket of size 10 should actually result in a remaining tokens value of -90 🤷 In any case, I'm almost certain this is not used by any system.

Reply:
> it's a little bit weird that a bucket can have a negative value

100%.  So... I looked at this and thought "huh... might be able to align the `take` and `put` behaviors around negative values, because IMO negative values are nonsensical for the idea of a "bucket".  Negative values effectively just become bottomless buckets.

So I attempted to see what the `take` behavior is... and it appears that it's just another strange bug.  It correctly will not go below zero more tokens are requested WHEN A DB RECORD ALREADY EXISTS.   But... when a DB record does NOT exist, and more tokens are requested than are present in the bucket, it correctly returns the fact that the result is non-conformant.  However, it instead uses the bucket size limit to create the record, erroneously leading to a full bucket state 😱 .

See https://github.com/josh-cain/limitd-redis/commit/ca7d25213ca70ab4598e1be9c2d02ed75386daba.